gemma2/gemma2_predict_gpu.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START generativeaionvertexai_gemma2_predict_gpu]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+	"github.com/googleapis/gax-go/v2"
+
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type ClientInterface interface {
```

Comment:
Before moving forward with this PR, we need to figure out a way to not include the interface in the sample - it's used only for testing purposes and shouldn't overcomplicate the sample code.

Reply:
@telpirion WDYT? Can we merge or would recommend additional changes?

Reply:
I've rewritten this sample to more closely match samples style. I'll approve and merge shortly.